compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5956,22 +5956,31 @@ public void visit(BLangXMLQName bLangXMLQName, AnalyzerData data) {
     private BSymbol findXMLNamespaceFromPackageConst(String localname, String prefix,
                                                      BPackageSymbol pkgSymbol, Location pos, AnalyzerData data) {
         // Resolve a const from module scope.
-        BSymbol constSymbol = symResolver.lookupMemberSymbol(pos, pkgSymbol.scope, data.env,
-                Names.fromString(localname), SymTag.CONSTANT);
+        BSymbol constSymbol =
+                symResolver.lookupPossibleMemberSymbol(pkgSymbol.scope, names.fromString(localname), SymTag.CONSTANT);
         if (constSymbol == symTable.notFoundSymbol) {
             if (!missingNodesHelper.isMissingNode(prefix) && !missingNodesHelper.isMissingNode(localname)) {
-                dlog.error(pos, DiagnosticErrorCode.UNDEFINED_SYMBOL, prefix + ":" + localname);
+                dlog.error(pos, DiagnosticErrorCode.UNDEFINED_CONSTANT_SYMBOL, prefix + ":" + localname);
             }
             return null;
         }
 
+        if (!data.env.enclPkg.packageID.equals(pkgSymbol.pkgID) && !Symbols.isPublic(constSymbol)) {
+            dlog.error(pos, DiagnosticErrorCode.ATTEMPT_REFER_NON_ACCESSIBLE_SYMBOL, constSymbol.name);
+            return null;
+        }
+
         // If Resolved const is not a string, it is an error.
         BConstantSymbol constantSymbol = (BConstantSymbol) constSymbol;
         if (constantSymbol.literalType.tag != TypeTags.STRING) {
             dlog.error(pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, symTable.stringType, constantSymbol.literalType);
             return null;
         }
 
+        if (!PackageID.isLangLibPackageID(pkgSymbol.pkgID)) {
```

Comment:
Why do we need to special case langlib?

Reply:
This was because this function being used for xml attribute access as well as generating xmlnamespace symbols for prefix based names. Thus refactored the code